src/safeStringify.ts

```diff
@@ -9,24 +9,34 @@
 
 import stringify from 'fast-safe-stringify'
 
+type ReplacerFn = (this: any, key: string, value: any) => any
+
 /**
- * Replacer to remove Circular values all together
+ * Replacer to handle bigints and remove Circular values all together
  */
-function replacer(_: any, value: any) {
-	if (value === '[Circular]') {
-		return
-	}
+function jsonStringifyReplacer(replacer?: ReplacerFn, removeCircular?: boolean): ReplacerFn {
+	return function (key, value) {
+		if (removeCircular && value === '[Circular]') {
+			return
+		}
+
+		const val = replacer ? replacer.call(this, key, value) : value
 
-	return value
+		if (typeof val === 'bigint') {
+			return val.toString()
+		}
+
+		return val
+	}
 }
 
 /**
  * Safely stringifies a Javascript native value
  */
-export function safeStringify(value: any): string {
+export function safeStringify(value: any, replacer?: ReplacerFn, space?: string | number): string {
 	try {
-		return JSON.stringify(value)
+		return JSON.stringify(value, jsonStringifyReplacer(replacer, false), space)
```

Comment:
I think we can get rid of `JSON.stringify` all together and use `stringify` method directly, since we passing a replacer in both cases anyways.

I used `try/catch` coz adding a replacer slows things down a bit and in 99% of cases there is no Circular reference, so the code was optimised for the majority use case

Reply:
Replacer is not the only reason of lower performance I think. Because `stringify` needs to loop over object and maintain seen objects to detect circular references. I will use it directly but maybe we can implement removing circular references ourselves using replacer and WeakSet and it will be more performant. Will look into it.